commands/peers/addpeer.go

```diff
@@ -43,11 +50,71 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		rest.SendHTTPError(w, http.StatusInternalServerError, *rsp.OpError)
 		return
 	}
+
+	h, _ := os.Hostname()
+	c, err := etcdclient.New(etcdclient.Config{Endpoints: []string{"http://" + h + ":2379"}})
```

Comment:
As @kshlm pointed out earlier that we'd need to store etcd client's globally and create it once at the life time of GlusterD such that we don't have to create it for each request.